### ISSUE TITLE
Step 7 Max Shared Hs Locus [24]

### ISSUE THREAD

**Issue by isoldevanriemsdijk** [dereneaton/ipyrad#260]
I'm working on a ddRADseq assembly with iPyRAD (v. 0.7.3), and I'm trying some different settings concerning heterozygosity. When I try to alter the parameter Max_Shared_Hs_Locus [24] in the params file to .75 or .9 and run step 7, the output files are (exactly) the same as for the default setting 0.5, whilst I'd expected it to retain more loci and/or detect more heterozygotes with a higher setting for this param.
My data consists of about 40 pure individuals of two different species, and about 350 hybrids (but not necessarily F1s). Therefore I expect pretty divergent sequences and many heterozygotes. I know this setting in particular is also meant to filter out paralogs (and .9 is probably a very bad idea). I have never worked with ipyrad/ddRAD data before, so my assumptions about changing this parameter could be entirely wrong.
Do you think nothing is changing in my output, because the number of heterozygous sites is never exceeding 50%? Or could something else be going on?

Another matter concerning Max_Shared_Hs_Locus [24], and possibly related, is when I try using two settings in the style of PE (so for example "0.5, 0.5" or "200, 200") iPyRAD gives an error, whilst my data is PE and all other settings are PE as well:

Error setting parameter 'max_shared_Hs_locus'
max_shared_Hs_locus must be int or float, you put: 0.75, 0.75
You entered: 0.75, 0.75

It does run when I use one setting in SE style ("0.75" or "200").
Does iPyRAD possibly not apply this setting to the R2 part of the loci/cluster?

Any suggestions or thoughts on the appropriate settings for this project are also very welcome!

Params file to reconstruct error (remove ", 0.75" from [24] to recreate step 7 which does not generate different data):
[params-mindepth_6_mxShHs_75.txt](https://github.com/dereneaton/ipyrad/files/1206433/params-mindepth_6_mxShHs_75.txt)

**Comment by isaacovercast** [comment]
Ok, well at least one mystery is solved. I looked at the code and really the code only accepts one parameter for PE and applies this setting to the whole locus. I will update the docs to reflect this.

**Comment by isaacovercast** [comment]
I just tested this parameter to be sure it's working as advertised and it does, so this does indeed mean the number of shared hets in your data never exceeds 50%.

**Comment by isoldevanriemsdijk** [comment]
Thank you for your help Isaac. The parameter file I used for this specific dataset (included in first post) appears to result in a reliable dataset. I had included some duplicate samples (made multiple libraries in the lab for the same individual), to see if different settings result in different SNP calling of heterozygotes, but all differences within individuals (~ 0.5 % of the SNPs in the final dataset) are caused by low sequence depth rather than this setting of Max_Shared_Hs_Locus [24]. And most of those differences within individuals are (indeed) calling a heterozygote as a homozygote, which for my analyses means, I've slightly underestimated the number of heterozygotes. Which I think is OK.
